docs/metadata.md

```diff
@@ -0,0 +1,166 @@
+---
+id: metadata
+title: Graphback Supported Metadata
```

Comment:
WOW. You are hero! This is the API reference we were all looking to implement. 

Thank you so much for doing it. I would probably put that under reference chapter etc.

CC @machi1990 @craicoverflow

Reply:
Hmm..I dunno about putting all metadata reference into one combined place. I believe they should be documented grouped with their purpose.

As an example. This page now documents (really nicely) the `@model` annotation. What happens to the `@model` annotation descriptions in [docs/intro/datamodel.md](https://github.com/aerogear/graphback/tree/master/docs/intro/datamodel.md)?

Do we maintain now maintain both? Or give a link from Data Model to this page (meaning more jumping around pages)

Reply:
Certainly, linking them up would be better (either from or to Data Model), but I think either way it would be best to have a single place that at least lists all supported annotations, wdyt, also thanks for pointing out, I will add the relationship annotations later today

Reply:
A link from https://github.com/aerogear/graphback/tree/master/docs/intro/datamodel.md to this part would be good. It should be fine too since this part now details nicely the `@model` annotation.